packages/mui-joy/src/styles/styleFunctionSx.spec.tsx

```diff
@@ -1,8 +1,8 @@
-import { experimental_sx as sx, styled } from '@mui/joy/styles';
+import { styled } from '@mui/joy/styles';
 
 // Can be used in the styled() utility
-const Test = styled('div')(
-  sx({
+const Test = styled('div')(({ theme }) =>
```

Comment:
Can you add more tests? I am curious if the `CSSObject` of `theme.unstable_sx(...)` works with all scenario.

1. using it with array
```js
const Test = styled('div')([
  ({ theme }) => theme.unstable_sx(...)
])
```

2. with pseudo class
```js
const Test = styled('div')({
  theme.unstable_sx(...),
  '&:hover': theme.unstable_sx(...),
})
```

Reply:
Added tests